engine/src/test/java/org/camunda/bpm/engine/test/history/HistoricProcessInstanceTest.java

```diff
@@ -2071,8 +2121,7 @@ public void shouldQueryByActivityIdInWithMultipleScope() {
     assertThat(queryByInnerServiceActivityId.size()).isEqualTo(1);
     assertThat(queryByInnerServiceActivityId.get(0).getId()).isEqualTo(processInstance.getId());
 
-    assertThat(queryBySubProcessActivityId.size()).isEqualTo(1);
-    assertThat(queryBySubProcessActivityId.get(0).getId()).isEqualTo(processInstance.getId());
+    assertNull(queryBySubProcessActivityId);
```

Comment:
We have 2 instances one at user task `innerTask` and one in user task `outerTask`. 
Querying by `subProcess` should return null. The Runtime query has the same behaviur (not returning instances by `subProcess` activityId)